src/lib/components/forms/form/form-components/suggestion/suggestion.component.spec.ts

```diff
@@ -1,24 +1,147 @@
+
         fixture.detectChanges();
     });
 
     it('should create', () => {
         expect(component).toBeTruthy();
+        expect(autoComplete).toBeTruthy();
+    });
+
+    it('should have #name as label of the input', async () => {
+        const formField: MatFormFieldHarness = await loader.getHarness(MatFormFieldHarness);
+        expect(await formField.getLabel()).toBe('');
+
+        component.name = 'My Label';
+        fixture.detectChanges();
+        expect(await formField.getLabel()).toBe(component.name);
+    });
+
+    it('should set required validation when #isRequired is set', async () => {
+        expect(await input.isRequired()).toBe(false);
+        component.isRequired = true;
+        fixture.detectChanges();
+
+        expect(await input.isRequired()).toBe(true);
+    });
+
+    it('should disable the input when #isDisabled is set', async () => {
+        expect(await autoComplete.isDisabled()).toBe(false);
+        expect(await input.isDisabled()).toBe(false);
+        component.isDisabled = true;
+        fixture.detectChanges();
+
+        expect(await autoComplete.isDisabled()).toBe(true);
+        expect(await input.isDisabled()).toBe(true);
+    });
+
+    describe('with tooltip', () => {
+        let tooltip: MockTooltipDirective;
+        beforeEach(() => {
+            tooltip = fixture.debugElement.query(By.directive(MockTooltipDirective)).injector.get(MockTooltipDirective);
+        });
+
+        it('should apply the tcTooltip directive to the form field with the given #tooltipText', () => {
+            expect(tooltip).toBeTruthy();
+            expect(tooltip.tcTooltip).toBe('');
+
+            component.tooltipText = 'My tooltip';
+            fixture.detectChanges();
+
+            expect(tooltip.tcTooltip).toBe(component.tooltipText);
+        });
+
+        it(`should set the tooltip's placement according to the input #tooltipPlacement`, () => {
+            expect(tooltip.placement).toBe(TerraPlacementEnum.TOP);
+
+            component.tooltipPlacement = TerraPlacementEnum.BOTTOM;
+            fixture.detectChanges();
+
+            expect(tooltip.placement).toBe(component.tooltipPlacement);
+        });
+    });
+
+    it('should update the value of the input when writing a new value via `writeValue()`', async () => {
+        component.listBoxValues = suggestionOptions;
+        component.writeValue(suggestionOption1.value);
+
+        fixture.detectChanges();
+        expect(await autoComplete.getValue()).toEqual(suggestionOption1.value.toString());
+        expect(await input.getValue()).toEqual(suggestionOption1.value.toString());
+    });
+
+    it('should call registered change callback whenever the value of the input is changed by the user', async () => {
+        const onChangeCallback: jasmine.Spy = jasmine.createSpy('onChange');
+        component.registerOnChange(onChangeCallback);
+
+        await input.setValue(suggestionOption1.caption);
```

Comment:
ich denke das wird so nicht funktionieren. Du musst stattdessen eine option selektieren (siehe MatAutocompleteHarness::selectOption())

Reply:
Wenn der Nutzer nämlich was in das input eintippt, sollte sich stattdessen die Liste der Vorschläge reduzieren.

Reply:
```suggestion
        await autoComplete.selectOption({ text: suggestionOption1.caption });
```